inc/components/class-parsely.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Class file for the Parsely component.
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving\Component;
+
+/**
+ * Defines the Parsely component.
+ */
+class Parsely extends Component {
+
+	/**
+	 * Name of site option associated with parsely
+	 *
+	 * @var string
+	 */
+	public static $option_field = 'parsely_site';
+
+	/**
+	 * Unique component slug.
+	 *
+	 * @var string
+	 */
+	public $name = 'parsely';
+
+	/**
+	 * Define the default config for Parsely
+	 *
+	 * @return array Default config values for this component.
+	 */
+	public function default_config() {
+		return [
+			'site' => $this->get_parsely_site(),
+		];
+	}
+
+	/**
+	 * Initialize parsely
+	 */
+	public function setup_head() {
+		add_action( 'wp_irving_head', [ $this, 'setup_parsely_meta' ], 10, 2 );
+	}
+
+	/**
+	 * Get Forum shortname for Parsely embeds
+	 *
+	 * @return array Filtered value for disqus forum shortname
+	 */
+	public function get_parsely_site() {
+		return apply_filters( 'wp_irving_parsely_site', get_option( self::$option_field ) );
+	}
+
+	/**
+	 * Add Parsely meta tags.
+	 *
+	 * @param \WP_Irving\Component\Head $head     Head object.
+	 * @param \WP_Query                 $wp_query WP_Query object.
+	 */
+	public function setup_parsely_meta( $head, $wp_query ) {
+		// Add Parsely meta tags.
+		if ( $wp_query->is_single() && ! empty( $wp_query->post ) ) {
+
+			// Get meta values.
+			$parsely_meta = $this->get_parsely_meta( $wp_query->post );
+
+			// Apply meta values.
+			foreach ( $parsely_meta as $name => $content ) {
+				$head->add_tag(
+					'meta',
+					[
+						'name'    => $name,
+						'content' => $content,
+					]
+				);
+			}
+		}
+	}
+
+	/**
+	 * Helper function that returns all the attributes needed for Parsely on a
+	 * single article.
+	 *
+	 * @param  \WP_Post $post Post object.
+	 * @return array
+	 */
+	public function get_parsely_meta( \WP_Post $post ) : array {
+
+		// Base properties.
+		$meta['parsely-link']    = get_the_permalink( $post );
```

Comment:
just curious, why declare it this way instead of `$meta = [ 'parsely-link' => /* val*/, /* ... etc. */ ]`

Reply:
Good question, I copypasta'd this code and mostly focused on removing unnecessary things. I agree with you though, will fix!